include/nbl/builtin/hlsl/device_capabilities_traits.hlsl

```diff
@@ -25,37 +23,16 @@ namespace hlsl
 
 namespace impl
 {
-NBL_GENERATE_GET_OR_DEFAULT(shaderFloat64, bool, false);
-NBL_GENERATE_GET_OR_DEFAULT(subgroupArithmetic, bool, false);
-NBL_GENERATE_GET_OR_DEFAULT(fragmentShaderPixelInterlock, bool, false);
-NBL_GENERATE_GET_OR_DEFAULT(maxOptimallyResidentWorkgroupInvocations, uint16_t, 0);
-NBL_GENERATE_GET_OR_DEFAULT(maxResidentInvocations, uint32_t, 0);
+    #include "nbl/video/device_capabilities_traits_defaults.h"
 }
 
-
 template<typename device_capabilities>
 struct device_capabilities_traits
 {
-    uint32_t computeOptimalPersistentWorkgroupDispatchSize(const uint32_t elementCount, const uint32_t workgroupSize, const uint32_t workgroupSpinningProtection)
-    {
-	    const uint32_t infinitelyWideDeviceWGCount = (elementCount - 1u) / (workgroupSize * workgroupSpinningProtection) + 1u; // need thtat divAndRoundUp
-	    return min(infinitelyWideDeviceWGCount,maxResidentInvocations/maxOptimallyResidentWorkgroupInvocations);
-    }
-    uint32_t computeOptimalPersistentWorkgroupDispatchSize(const uint32_t elementCount, const uint32_t workgroupSize)
-    {
-	    return computeOptimalPersistentWorkgroupDispatchSize(elementCount,workgroupSize,1u);
-    }
```

Comment:
leave this in, this is nice and readable doesn't need autogenning

Reply:
Done